drivers/volume/portworx/portworx.go

```diff
@@ -1775,6 +1888,65 @@ func (d *portworx) CollectDiags(n node.Node) error {
 	return nil
 }
 
+// CalculateTotalSize calculates expected size based on autopilot rule, initial and workload sizes
+func (d *portworx) CalculateTotalSize(apRule apapi.AutopilotRule, initSize, wklSize uint64) uint64 {
+	var maxSize, calculatedTotalSize uint64
+	calculatedTotalSize = initSize
+	applyRuleTrigger := true
+	for applyRuleTrigger {
```

Comment:
the logic here is little difficult to understand. Can you add a summary of what this is doing?